src/lib/state/actions/deposit.js

```diff
@@ -193,13 +194,50 @@ export const submitReview = (draft, { reviewComment }) => {
     try {
       const reqResponse = await config.service.drafts.submitReview(
         draftWithLinks.links,
-        reviewComment
+        reviewComment,
+        directPublish
       );
       const request = reqResponse.data;
-      // after submitting for review, redirect to the review record
-      const rawRequestURL = config.config.links.user_dashboard_request;
-      const requestURL = rawRequestURL.replace("<request_pid_value>", request.id);
-      window.location.replace(requestURL);
+      if (directPublish) {
+        // after publishing, redirect to the published record
+        const recordURL = draftWithLinks.links.record_html;
+        window.location.replace(recordURL);
+      } else {
+        // after submitting for review, redirect to the review record
+        const rawRequestURL = config.config.links.user_dashboard_request;
+        const requestURL = rawRequestURL.replace("<request_pid_value>", request.id);
+        window.location.replace(requestURL);
+      }
+    } catch (error) {
+      dispatch({
+        type: DRAFT_SUBMIT_REVIEW_FAILED,
+        payload: { errors: error.errors },
+      });
+      throw error;
+    }
+  };
+};
+
+export const submitAndPublishReview = (draft) => {
+  return async (dispatch, getState, config) => {
+    // TODO: add me back
```

Comment:
minor: sneaky TODO commented

Reply:
that function needs to be removed now :)